js/app.jsx

```diff
@@ -74,54 +72,57 @@ var SearchableRemoteListMixin = {
   //need to implement `function buildURL(query) {...}`
   loading: {meta:{}, objects: [{name:"Loading..."}]},
   notFound: {data:{objects:[{name:"Can't find anything. Try loosen the search criteria."}]}},
-  loadRemoteData: function(url) {
-    $.ajax({
-      url: url,
-      timeout: 10000, // Force trigger the error callback
+  
+  updateListUIState: function(data) {
+    // 1. Hightlight the loaded page.
+    $('li.pre-active').attr('class', 'active'); 
 
-      success: function(data) {
-        if (data.meta.next == null || typeof data.meta.next == 'undefined') 
-          this.setState({data: data, hasNoLinkToShow: true});
-        else
-          this.setState({data: data, hasNoLinkToShow: false});
-      }.bind(this),
+    // 2. Unchecked all boxes.
+    $.each($('tr td input[type=checkbox]:checked'), function() {
+      $(this).prop('checked', false);
+    });
 
-      /*
-      FIXME: this doesn't seem to work under jsonp proxy
-      statusCode: {
-        400: function() {
-          alert.log('bad request'); }, },
-      */
-      error: function(xhr, status, err) {
-        this.setState(this.notFound)
-        console.error(xhr, status, err.toString());
-      }.bind(this)
+    // 3. Check boxes whose ID matches checkedIDs.
+    var checkedIDs = this.state['checked'];
+    $.each($('tr td input[type=checkbox]'), 
+      function() { 
+        for (i = 0; i < checkedIDs.length; i++) { 
+          if ($(this).val() == checkedIDs[i])
+            $(this).prop('checked', true);
+        }
     });
-  },
-  //FIXME: Change this to pagination
-  loadMore: function() {
-    var url = config.baseUrl + this.state.data.meta.next;
-    //console.log(this.state.data.meta)
-    //console.log(url)
+  }, 
+
+  loadOnePage: function(url, clickedPageNum) {
+    clickedPageNum = typeof clickedPageNum !== 'undefined' ? clickedPageNum : 1;
+    var selectedPageOffset = (clickedPageNum-1) * this.state.data.meta.limit;
+    url = url.replace(/offset=[0-9]+/, "offset=" + selectedPageOffset);
+
+    console.log("===> loadOnePage URL = ", url)
+
     $.ajax({
       url: url,
+      timeout: 10000, // Force trigger the error callback
```

Comment:
Not sure if we still need this right now. It was here because the JSONP does not return 404 or other errors. I guess the default timeout would be suffice.